lychee-lib/src/filter/mod.rs

```diff
@@ -49,11 +49,11 @@ impl Filter {
     #[must_use]
     /// Whether e-mails aren't checked
     pub fn is_mail_excluded(&self, uri: &Uri) -> bool {
-        uri.is_mail() && self.exclude_mail
+        self.exclude_mail && uri.is_mail()
```

Comment:
Conditions switched as I guess checking flag is much cheaper than checking whether the URI is an email address. This also matches the order in `is_ip_excluded` and the new `is_host_excluded` below.